src/arraystring.rs

```diff
@@ -1,11 +1,12 @@
 //! `ArrayString` definition and Api implementation
 
-use crate::utils::{encode_char_utf8_unchecked, is_char_boundary, is_inside_boundary, never};
+use crate::utils::{append_char_truncate, encode_char_utf8_unchecked, is_char_boundary, is_inside_boundary, never};
 use crate::utils::{shift_left_unchecked, shift_right_unchecked, truncate_str, IntoLossy};
 use crate::{error::Error, prelude::*};
 use core::char::{decode_utf16, REPLACEMENT_CHARACTER};
-use core::str::{from_utf8, from_utf8_unchecked};
+use core::str::from_utf8;
 use core::{cmp::min, ops::*, ptr::copy_nonoverlapping};
+use std::str::{from_utf8_unchecked, FromStr};
```

Comment:
std is an optional feature, so FromStr should only be implemented in that case. And for optional features we should be inlining the import in the usage.

Why did you move from_utf8_unchecked import for this?

Reply:
Didn't really pay attention to that, i didn't realize std was a separate feature